layouts/partials/nav.html

```diff
@@ -1,10 +1,10 @@
 <nav class="nav" role="navigation">
   <ul class="nav__list">
-    {{ $currentPage := . }} 
+    {{ $currentPage := . }}
     {{ range .Site.Menus.main }}
       <li>
-        <a {{if or ($currentPage.IsMenuCurrent "main" .) (and (eq .Name "blog") (eq $currentPage.Section "blog")) }} class="active"
-          {{end}} href="{{ .URL }}">{{ .Name }}</a>
+        <a {{ if or ($currentPage.IsMenuCurrent "main" .) (eq $currentPage.Section .Identifier) }} class="active"
+        {{end}} href="{{ .URL }}">{{ .Name }}</a>
```

Comment:
Sorry I'm terrible with `go` syntax. What's the change here?

Reply:
I was using the solution listed here: https://discourse.gohugo.io/t/highlight-current-menu-item-config-toml/22723/7

Given this menu config:
```toml
[[menu.main]]
  identifier = "about"
  name = "about"
  title = "About"
  url = "/"

[[menu.main]]
  identifier = "blog"
  name = "blog"
  title = "Blog"
  url = "/blog"

```
Basically, the first conditional highlights the homepage (`url = "/"`), and the second conditional highlights whatever menu whose identifier is the same as the page's [section](https://gohugo.io/content-management/sections/#section-page-variables-and-methods) (the first path element in the directory, for example, `/blog/funny/mypost/ => blog`.)

Reply:
Awesome stuff, thanks for the explanation!